Lib/tokenize.py

```diff
@@ -95,6 +96,7 @@ def _all_string_prefixes():
                 result.add(''.join(u))
     return result
 
+@functools.lru_cache
```

Comment:
I think we should limit the max size of the cache in this call so this doesn't blow up for some super intensive cases.

Reply:
In the `re` module we use the same trick and we have 512 as the max size of the cache.

Reply:
I commented on the other PR, there's only 5 possible regex strings that go through this call

Reply:
Roger. Then is fine, although I would have preferred to manually pre-compile those and avoid importing functools

Reply:
that was my original patch, it increased import time of this module by ~20ms

Reply:
How much is importing `functools`? Also, we can still go that way if we do the compilation lazily, but I don't think is worth it.

Reply:
it is already imported via `tokenize => re => functools` so 0

Reply:
Oh, then I retract my concern. This is an absolutely better approach. Thanks for following with me :)